### ISSUE TITLE
Failed to find a valid OpenAIConfiguration, when using environment variables or .openai in home dir

### ISSUE THREAD

**Issue by pinkwerks** [RageAgainstThePixel/com.openai.unity#144]
# Bug Report
## Overview

I went with "use system environment variables" recommendation from the docs and set both OPENAI_API_KEY and OPENAI_ORGANIZATION_ID in my environment. (https://github.com/RageAgainstThePixel/com.openai.unity#use-system-environment-variables)

## To Reproduce

1. Loading "Chat" sample scene from package manager
2. Press play
3. I get the error "Failed to find a valid OpenAIConfiguration!'. 
4. To work around I had to create a OpenAIConfiguration.asset and fill it out

## Expected behavior
I'd expect to set the appropriate environment variables and not have to create the  OpenAIConfiguration.asset in my Unity project.

## Additional note
I also tried the adding the config file to my $env:USERPROFILE\.openai  and get the same error. It seems like its required to generate the OpenAIConfiguration.asset and fill it out.

**Comment by StephenHodgson** [comment]
I use environment variables for my production builds. Are you sure you restarted your computer after settings them?

**Comment by StephenHodgson** [comment]
I'll need more information about the system you're running on and how you're creating this file

**Comment by StephenHodgson** [comment]
but at first glance, this doesn't look correct to me `$env:USERPROFILE.openai`

**Comment by pinkwerks** [comment]
Sorry it should have been $env:USERPROFILE\.openai - forgot the slash. (my home directory on my windows 10 box) I created it with visual studio code. 

![image](https://github.com/RageAgainstThePixel/com.openai.unity/assets/538256/d398059a-4284-4f7e-9b05-4a4568e17dc4)

Before I filed the bug, or tried the ~/.openai file, I was able to attach a debugger to Unity and verify that the environment variable was being read correctly with `Environment.GetEnvironmentVariable()` - yet I still got the error about the config asset.

However, now that I'm trying to repo again, after deleting my OpenAIConfiguration.asset and the ~/.openai file. It appears to work. I'm confused. Maybe some first run thing or caching? Feel free to close since I can't repo, Sorry, thanks for your time and putting this package together.
